Sources/web3swift/EthereumAPICalls/Ethereum/Eth+GetTransactionDetails.swift

```diff
@@ -8,9 +8,8 @@ import BigInt
 import Core
 
 extension Web3.Eth {
-    public func transactionDetails(_ txhash: Data) async throws -> TransactionDetails {
-        guard let hexString = String(data: txhash, encoding: .utf8)?.addHexPrefix() else { throw Web3Error.dataError }
-        let requestCall: APIRequest = .getTransactionByHash(hexString)
-        return try await APIRequest.sendRequest(with: self.provider, for: requestCall).result
+    public func transactionDetails(_ txHash: Data) async throws -> TransactionDetails {
+        try await APIRequest.sendRequest(with: self.provider,
+                                         for: .getTransactionByHash(txHash.toHexString().addHexPrefix())).result
```

Comment:
I'd keep the extra line for the request call for readability and easier debugging but that's personal preference.

Reply:
The point about debugging is actually a good one.
Will revert it back.